src/ripple/app/tx/impl/SusPay.cpp

```diff
@@ -103,8 +101,10 @@ namespace ripple {
         If the SusPay ledger entry specifies sfCancelAfter, the
         transaction will fail if sfCancelAfter <= parentCloseTime.
 
-        NOTE: It must always be possible to verify the condition
-              without retrieving the SusPay ledger entry.
+        NOTE: The reason the condition must be specified again
+              is because it must always be possible to verify
+              the condition without retrieving the SusPay
+              ledger entry.
```

Comment:
I don't really see the point of doing a `validateTrigger` that way (except that maybe you verify that the fulfillment is for an empty slice, which is a requirement for the trigger).

You also have to deserialize the fulfillment twice and generate the condition out of the fulfillment not once but twice (one in `preflight` and one in `doApply` and that operation may require multiple SHA256 operations, followed by one serialization or deserialization.

All this to save from having to specify the condition again? The submitter has the fulfillment (they need to supply it) so they can always generate the condition.